BSM guide

hillaryfraley opened the pull request:
## Description
Creates an initial BSM guide, with a component and rule template examples for a business service scenario. Also includes examples for different BSM approaches (sampling, polling, hybrid).

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2956

hillaryfraley committed:
BSM guide

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-management.md:
```diff
@@ -0,0 +1,243 @@
+Although you can configure components and rule templates within the web UI, both are Sensu resources with complete definitions.
+
+Here is an example component definition that references the `status-threshold` rule template:
+
```
These examples should be replaced with examples to fit the company website scenario described in the introduction.

hillaryfraley committed:
Update to use service components

hillaryfraley commented:
A couple notes from the BSM sheet that may be helpful:

- "One-click silence (or acknowledge) of BSM alerts via Slack, email, or PagerDuty." Sean clarifies: Need to add some clarification, you cannot silence the event from these services, but you can silence them in Sensu itself and further notifications will not be sent to the service.
- BSM "...effectively replaces dependencies. A better way to solve the problem."
- Dynamic BSM Components: In addition to individually selecting the tests that make up a Business Service Component (which exists as a static list), you can use Infrastructure-as-code workflows to create dynamic Business Services. Use labels and/or names (or other entity fields) to automatically identify events of a Business Service Component as they are added or removed.
- Reduce Alert Fatigue: Avoid false positives due to scheduled maintenance of underlying entities (or tests). You can set up a delay before getting notified about a Business Service failure to only get notified when a Business Service state persists for some time OR after a number of polling intervals. You can also set up escalation of an event (e.g., immediately open a ticket in ServiceNow when a Business Service goes down; if it’s still down after 5 minutes, the service owner is paged; after 10 minutes, the service owner’s manager is paged).
- A user can query which components of a business service is causing it to be in a non-ok state. This detail can be transmitted to the handler such as ServiceNow or Slack, and displayed in the UI and queried using the Sensu API.
- Auto-Remediation: You can query the status of the Business Service Entity using sensuctl or API to tie into an auto-remediation workflow.
- You can do a reverse query using the API to determine which Business Service Entity an event belongs to, so that you can notify Business Service owners proactively before you take an underlying entity down for maintenance.

Notes from demo:
- Many users will not need to create their own rule templates. Instead, they are more likely to apply our preconfigured rule templates for specific cases by referencing them in their component definitions.

hillaryfraley committed:
Rename file and update intro explanation

hillaryfraley committed:
A few edits and added web UI and APIs sections

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,264 @@
+[2]: ../rule-templates/
+[3]: ../../web-ui/bsm-module/
+[4]: #example-service-component
+[5]: ../../sensuctl/
```
```suggestion
[5]: ../../sensuctl/
[6]: ../../commercial/
```

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,264 @@
+  sensu-go-6.3:
+    parent: observe-schedule
+---
+
```
```suggestion

**COMMERCIAL FEATURE**: Access business service monitoring (BSM) in the packaged Sensu Go distribution.
For more information, see [Get started with commercial features][6].
```

hillaryfraley committed:
Add commercial feature notice

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,267 @@
+
+**COMMERCIAL FEATURE**: Access business service monitoring (BSM) in the packaged Sensu Go distribution.
+For more information, see [Get started with commercial features][6].
+Sensu's business service monitoring (BSM) provides high-level visibility into the current health of any number of your business services.
```
```suggestion

Sensu's business service monitoring (BSM) provides high-level visibility into the current health of any number of your business services.
```

hillaryfraley committed:
add missing line

hillaryfraley committed:
Merge branch 'master' into 6.3.0/bsm-guide

hillaryfraley committed:
Update service component example

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,271 @@
+To continue the company website example, if the primary webserver fails but the backup webserver does not, you might use a rule template that creates a service ticket to address the next workday (in addition to the rule template that is emitting "OK" events for the current status page).
+Another monitoring rule might trigger an alert to the on-call operator should both webservers or the inventory database fail.
+
+## Example service component
```
```suggestion
## Service component example
```

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,271 @@
+
+{{< /language-toggle >}}
+
+## Example rule template
```
```suggestion
## Rule template example
```

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,271 @@
+Service components are the elements that make up your business services.
+Rule templates define the monitoring rules that produce events for service components based on customized evaluation expressions.
+
+## Use case example
```
```suggestion
```

hillaryfraley committed:
Update headings

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,269 @@
+
+BSM service components and rule templates are Sensu resources with complete definitions, so you can use Sensu's [service component][1] and [rule template][2] APIs to create, retrieve, update, and delete service components and rule templates.
+
+You can use [sensuctl][5] to create and manage service components and rule templates via the APIs from the command line.
```
```suggestion
You can also use [sensuctl][5] to create and manage service components and rule templates via the APIs from the command line.
```

hillaryfraley committed:
Add "also"

hillaryfraley committed:
Merge branch 'master' into 6.3.0/bsm-guide

hillaryfraley committed:
Add BSM public preview note

jamesdphillips commented:
I could have missed it, but we should call out that the feature is only available to those using the PostgreSQL adapter.

hillaryfraley committed:
Merge branch 'master' into 6.3.0/bsm-guide

hillaryfraley committed:
Add note about postgres req and remove hybrid approach

hillaryfraley commented:
I removed the hybrid content and added a note re: PostgreSQL requirement. The only outstanding item aside from overall review is where I have placeholders for sampling and polling examples. Do we have examples of each?

hillaryfraley committed:
Merge branch 'master' into 6.3.0/bsm-guide

echlebek commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,262 @@
+
+#### Polling example
+
+...
```
I think we can simply drop this text. It's incorrect, since BSM was not implemented this way.

hillaryfraley replied:
```suggestion
```

hillaryfraley committed:
Remove section on sampling and polling approaches

hillaryfraley commented on content/sensu-go/6.3/observability-pipeline/observe-schedule/business-service-monitoring.md:
```diff
@@ -0,0 +1,231 @@
+
+You can also use [sensuctl][5] to create and manage service components and rule templates via the APIs from the command line.
+
+
```
```suggestion
```

hillaryfraley committed:
Remove stray line

hillaryfraley committed:
Merge branch 'master' into 6.3.0/bsm-guide
